apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTask.java

```diff
@@ -140,42 +152,43 @@ private Optional<JobState> submitJob(Attributes typeAttributes) {
 
   private Optional<JobState> pollJobStatus(Attributes typeAttributes) {
     long startTime = System.currentTimeMillis();
-    while (!jobFinished()) {
-      long elapsedTime = System.currentTimeMillis() - startTime;
-      if (elapsedTime > timeoutMs) {
+    try {
+      while (!jobFinished()) {
+        long elapsedTime = System.currentTimeMillis() - startTime;
         Optional<JobState> job = jobsClient.getState(jobId);
```

Comment:
jobFinished is also invoking jobsClient.getState. So we should avoid invoking twice. We can refactor the code and change the while loop to do while and has finished can be checked later based on already retrieved jobState.